src/livestreamer_cli/main.py

```diff
@@ -444,9 +449,12 @@ def format_valid_streams(streams):
             joined = delimiter.join(synonyms)
             name = "{0} ({1})".format(name, joined)
 
-        validstreams.append(name)
+        validstreams.append((name, weight))
+
+    validstreams.sort(key=lambda stream: stream[1])
```

Comment:
Well that's just embarrassing. Literally in the middle of the function I was editing.
Thanks.